pulsar-zookeeper-utils/src/main/java/com/yahoo/pulsar/zookeeper/ZooKeeperCache.java

```diff
@@ -74,11 +79,13 @@
     protected final Cache<String, Set<String>> childrenCache;
     protected final Cache<String, Boolean> existsCache;
     protected final OrderedSafeExecutor executor;
+    protected final ScheduledExecutorService scheduledExecutor;
 
     protected AtomicReference<ZooKeeper> zkSession = new AtomicReference<ZooKeeper>(null);
 
-    public ZooKeeperCache(ZooKeeper zkSession, OrderedSafeExecutor executor) {
+    public ZooKeeperCache(ZooKeeper zkSession, OrderedSafeExecutor executor, ScheduledExecutorService scheduledExecutor) {
```

Comment:
`ZooKeeperCache(zkSession, executor, scheduledExecutor)` being used by `GlobalZooKeeperCache` and `LocalZKCache` so, I thought it would be better to share `executor` created by `PulsarService`.

However, `ZooKeeperCache(ZooKeeper zkSession)` is being used by `BookKeeperClientFactoryImpl` so, it can maintain ZKCache life-cycle by calling `stop` method.

>Can we just force to use the internal executor? 

then for every ZkCache object it creates separate thread-pool and what should be the default thread-number in the pool?

>If someone is passing around other executors, they might get shut down when we close the ZooKeeperCache.

then should we keep a flag if `ZooKeeperCache` has initialized its own executor and close only if it owns?

So, should we introduce a flag or we should create a thread-pool for each ZKCache instance and with what size?